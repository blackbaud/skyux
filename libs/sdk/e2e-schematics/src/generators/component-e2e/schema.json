{
  "$schema": "http://json-schema.org/schema",
  "$id": "library-e2e",
  "title": "Create a library e2e",
  "description": "Creates an e2e infrastructure for a library.",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the library.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What is the name of the library?",
      "pattern": "^[a-zA-Z].*$"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the library (used for linting)."
    }
  },
  "additionalProperties": false,
  "required": ["name"]
}
