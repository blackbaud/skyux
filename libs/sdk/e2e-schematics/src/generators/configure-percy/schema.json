{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "configure-percy",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "e2e project name",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    },
    "skipFormat": {
      "type": "boolean",
      "aliases": ["skip-format"],
      "description": "Skip formatting files.",
      "default": false
    }
  }
}
