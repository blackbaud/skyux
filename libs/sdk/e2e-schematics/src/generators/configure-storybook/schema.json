{
  "$schema": "http://json-schema.org/schema",
  "$id": "configure-storybook",
  "title": "Configure Storybook",
  "description": "Adds configuration to Storybook.",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the project using Storybook.",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    }
  },
  "additionalProperties": false
}
