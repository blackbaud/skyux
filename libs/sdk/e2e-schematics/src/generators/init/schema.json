{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "Init",
  "title": "Configure E2E environment"
}
