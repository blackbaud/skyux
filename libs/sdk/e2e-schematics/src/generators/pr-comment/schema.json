{
  "$schema": "http://json-schema.org/schema",
  "$id": "PrComment",
  "title": "PR Comment",
  "description": "Generate markdown for a PR comment or a PR preview README.md",
  "type": "object",
  "properties": {
    "pr": {
      "type": "number",
      "description": "PR Number",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "Which PR is this for?"
    },
    "storybooks": {
      "type": "string",
      "description": "Comma separated list of storybooks to include in the comment",
      "alias": "s",
      "x-prompt": "Which storybook projects should be listed?"
    },
    "apps": {
      "type": "string",
      "description": "Comma separated list of apps to include in the comment",
      "alias": "a"
    }
  },
  "required": ["pr", "storybooks"]
}
