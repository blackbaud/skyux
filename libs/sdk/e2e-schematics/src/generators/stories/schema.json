{
  "$schema": "http://json-schema.org/schema",
  "$id": "Stories",
  "title": "Create Storybook stories/specs",
  "description": "Creates Storybook stories/specs for all Angular components declared in a project.",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What's the name of the project for which you want to generate stories?",
      "aliases": [
        "app",
        "storybook-app",
        "storybookapp",
        "storybookApp",
        "storybook"
      ]
    },
    "generateCypressSpecs": {
      "type": "boolean",
      "description": "Specifies whether to automatically generate `*.spec.ts` files in the Cypress e2e app generated by the `cypress-configure` generator.",
      "default": true
    },
    "cypressProject": {
      "type": "string",
      "description": "The Cypress project to generate the stories under. This is inferred from `name` by default."
    }
  },
  "additionalProperties": false,
  "required": ["project"]
}
