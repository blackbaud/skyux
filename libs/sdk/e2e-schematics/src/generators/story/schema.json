{
  "$schema": "http://json-schema.org/schema",
  "$id": "Story",
  "title": "Create a new story with a new component and corresponding tests",
  "description": "Create a new story with a new component and corresponding tests",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the component.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the new storybook component?"
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "x-prompt": "What's the name of the storybook application for which you want to generate the story?",
      "aliases": [
        "app",
        "storybook-app",
        "storybookapp",
        "storybookApp",
        "storybook"
      ]
    },
    "generateCypressSpecs": {
      "type": "boolean",
      "description": "Specifies whether to automatically generate `*.spec.ts` files in the Cypress e2e app generated by the `cypress-configure` generator.",
      "default": true
    },
    "cypressProject": {
      "type": "string",
      "description": "The Cypress project to generate the stories under. This is inferred from `project` by default."
    },
    "includeTests": {
      "type": "boolean",
      "description": "Include tests in the library.",
      "default": false
    }
  },
  "additionalProperties": false,
  "required": ["name", "project"]
}
