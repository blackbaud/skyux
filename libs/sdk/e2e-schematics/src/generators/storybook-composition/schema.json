{
  "$schema": "http://json-schema.org/schema",
  "$id": "storybook-composition",
  "type": "object",
  "properties": {
    "projectsJson": {
      "type": "string",
      "description": "JSON array of projects to include"
    },
    "baseUrl": {
      "type": "string",
      "description": "Web address containing published Storybooks."
    }
  },
  "required": ["projectsJson", "baseUrl"]
}
