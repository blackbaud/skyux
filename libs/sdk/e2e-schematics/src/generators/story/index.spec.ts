import {
  applicationGenerator,
  libraryGenerator,
  storybookConfigurationGenerator,
} from '@nrwl/angular/generators';
import { Tree } from '@nrwl/devkit';
import { createTreeWithEmptyV1Workspace } from '@nrwl/devkit/testing';
import { Linter } from '@nrwl/linter';
import { RoutingScope } from '@schematics/angular/module/schema';

import { angularModuleGenerator } from '../../utils';

import storyGenerator from './index';
import { ComponentGeneratorSchema } from './schema';

describe('component generator', () => {
  let appTree: Tree;
  let options: ComponentGeneratorSchema;

  beforeEach(async () => {
    appTree = createTreeWithEmptyV1Workspace();
    options = {
      name: 'example',
      project: 'test-storybook',
      generateCypressSpecs: true,
    };
    await applicationGenerator(appTree, {
      name: 'test',
    });
    await applicationGenerator(appTree, {
      name: 'test-storybook',
    });
    await libraryGenerator(appTree, {
      name: 'storybook',
      directory: 'components',
    });
    await storybookConfigurationGenerator(appTree, {
      name: 'test-storybook',
      generateCypressSpecs: false,
      configureCypress: true,
      linter: Linter.None,
      generateStories: false,
    });
    await angularModuleGenerator(appTree, {
      name: 'test-router',
      routing: true,
      routingScope: RoutingScope.Root,
      flat: true,
      project: 'test-storybook',
    });
  });

  it('should run successfully', async () => {
    await storyGenerator(appTree, options);
    expect(
      appTree.read(
        'apps/test-storybook/src/app/example/example.component.stories.ts',
        'utf-8'
      )
    ).toMatchSnapshot();
  });

  it('should run successfully with sub directory', async () => {
    await angularModuleGenerator(appTree, {
      name: 'test-sub',
      project: 'test-storybook',
      routing: true,
      routingScope: RoutingScope.Child,
      route: 'sub2',
      module: 'test-router',
    });
    await storyGenerator(appTree, {
      ...options,
      name: 'example/sub-example',
    });
    expect(
      appTree.read(
        'apps/test-storybook/src/app/example/sub-example/sub-example.component.stories.ts',
        'utf-8'
      )
    ).toMatchSnapshot();
  });

  it('should run successfully with --module', async () => {
    await storyGenerator(appTree, {
      ...options,
      module: 'test-router',
    });
    expect(
      appTree.read(
        'apps/test-storybook/src/app/example/example.component.stories.ts',
        'utf-8'
      )
    ).toMatchSnapshot();
  });

  it('should run successfully, finding a module', async () => {
    await angularModuleGenerator(appTree, {
      name: 'sub-test',
      project: 'test-storybook',
      flat: true,
      routing: true,
      routingScope: RoutingScope.Child,
      route: 'sub1',
      module: 'test-router',
    });
    await angularModuleGenerator(appTree, {
      name: 'test-sub',
      project: 'test-storybook',
      routing: true,
      routingScope: RoutingScope.Child,
      route: 'sub2',
      module: 'test-router',
    });
    await angularModuleGenerator(appTree, {
      name: 'test-sub/test-sub-sub',
      project: 'test-storybook',
      routing: true,
      routingScope: RoutingScope.Child,
      route: 'sub3',
      module: 'test-sub',
    });
    await storyGenerator(appTree, options);
    expect(
      appTree.read(
        'apps/test-storybook/src/app/example/example.component.stories.ts',
        'utf-8'
      )
    ).toMatchSnapshot();
    expect(
      appTree.read(
        'apps/test-storybook/src/app/example/example.component.ts',
        'utf-8'
      )
    ).toMatchSnapshot();
    expect(
      appTree.read(
        'apps/test-storybook/src/app/example/example.module.ts',
        'utf-8'
      )
    ).toMatchSnapshot();
    expect(
      appTree.read('apps/test-storybook/src/app/test-router.module.ts', 'utf-8')
    ).toMatchSnapshot();
  });

  it('should use -storybook project', async () => {
    await storyGenerator(appTree, {
      ...options,
      project: 'test',
    });
    expect(
      appTree.read(
        'apps/test-storybook/src/app/example/example.component.stories.ts',
        'utf-8'
      )
    ).toMatchSnapshot();
    // expect(appTree.listChanges().filter(c => c.path.includes('test-storybook')).map(c => c.path)).toEqual([]);
    expect(
      appTree.isFile(
        'apps/test-storybook-e2e/src/integration/example.component.spec.ts'
      )
    ).toBeTruthy();
    expect(
      appTree.read(
        'apps/test-storybook-e2e/src/integration/example.component.spec.ts',
        'utf-8'
      )
    ).toMatchSnapshot();
  });

  it('should throw errors', async () => {
    appTree.write('apps/test-storybook/src/app/example/example.ts', 'test');
    try {
      await storyGenerator(appTree, options);
      fail();
    } catch (e) {
      expect(e.message).toEqual(`example already exists for test-storybook`);
    }
    appTree.delete('apps/test-storybook/src/app/example/example.ts');
    try {
      await storyGenerator(appTree, {
        ...options,
        project: 'wrong',
      });
      fail();
    } catch (e) {
      expect((e as Error).message).toEqual('Unable to find project wrong');
    }
    await libraryGenerator(appTree, {
      name: 'wrong',
    });
    try {
      await storyGenerator(appTree, {
        ...options,
        project: 'wrong',
      });
      fail();
    } catch (e) {
      expect((e as Error).message).toEqual(
        'Unable to find project wrong-storybook'
      );
    }
    await applicationGenerator(appTree, {
      name: 'wrong-storybook',
    });
    try {
      await storyGenerator(appTree, {
        ...options,
        project: 'wrong',
      });
      fail();
    } catch (e) {
      expect((e as Error).message).toEqual(
        'Storybook is not configured for wrong-storybook'
      );
    }
    appTree.delete('apps/test-storybook/src/app/test-router-routing.module.ts');
    try {
      await storyGenerator(appTree, options);
      fail();
    } catch (e) {
      expect((e as Error).message).toEqual(
        'Could not find a router module to add the component to. Please specify a module using the --module option.'
      );
    }
  });
});
