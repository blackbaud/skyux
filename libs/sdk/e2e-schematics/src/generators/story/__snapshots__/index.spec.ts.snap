// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component generator should run successfully 1`] = `
"import { ExampleComponent } from './example.component';
import { ExampleModule } from './example.module';
import { Meta, moduleMetadata } from '@storybook/angular';

export default {
  id: 'examplecomponent-example',
  title: 'Components/Example',
  component: ExampleComponent,
  decorators: [
    moduleMetadata({
      imports: [ExampleModule],
    }),
  ],
} as Meta<ExampleComponent>;
export const Example = {
  render: (args: ExampleComponent) => ({
    props: args,
  }),
  args: {},
};
"
`;

exports[`component generator should run successfully with --module 1`] = `
"import { ExampleComponent } from './example.component';
import { ExampleModule } from './example.module';
import { Meta, moduleMetadata } from '@storybook/angular';

export default {
  id: 'examplecomponent-example',
  title: 'Components/Example',
  component: ExampleComponent,
  decorators: [
    moduleMetadata({
      imports: [ExampleModule],
    }),
  ],
} as Meta<ExampleComponent>;
export const Example = {
  render: (args: ExampleComponent) => ({
    props: args,
  }),
  args: {},
};
"
`;

exports[`component generator should run successfully with sub directory 1`] = `
"import { SubExampleComponent } from './sub-example.component';
import { SubExampleModule } from './sub-example.module';
import { Meta, moduleMetadata } from '@storybook/angular';

export default {
  id: 'subexamplecomponent-subexample',
  title: 'Components/Example/Sub Example',
  component: SubExampleComponent,
  decorators: [
    moduleMetadata({
      imports: [SubExampleModule],
    }),
  ],
} as Meta<SubExampleComponent>;
export const SubExample = {
  render: (args: SubExampleComponent) => ({
    props: args,
  }),
  args: {},
};
"
`;

exports[`component generator should run successfully, finding a module 1`] = `
"import { ExampleComponent } from './example.component';
import { ExampleModule } from './example.module';
import { Meta, moduleMetadata } from '@storybook/angular';

export default {
  id: 'examplecomponent-example',
  title: 'Components/Example',
  component: ExampleComponent,
  decorators: [
    moduleMetadata({
      imports: [ExampleModule],
    }),
  ],
} as Meta<ExampleComponent>;
export const Example = {
  render: (args: ExampleComponent) => ({
    props: args,
  }),
  args: {},
};
"
`;

exports[`component generator should run successfully, finding a module 2`] = `
"import { Component } from '@angular/core';

@Component({
  selector: 'proj-example',
  templateUrl: './example.component.html',
  styleUrls: ['./example.component.scss'],
})
export class ExampleComponent {}
"
`;

exports[`component generator should run successfully, finding a module 3`] = `
"import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ExampleRoutingModule } from './example-routing.module';
import { ExampleComponent } from './example.component';
@NgModule({
    declarations: [
        ExampleComponent
    ],
    imports: [
        CommonModule,
        ExampleRoutingModule
    ],
    exports: [ExampleComponent]
})
export class ExampleModule {
}
"
`;

exports[`component generator should run successfully, finding a module 4`] = `
"import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { TestRouterRoutingModule } from './test-router-routing.module';


@NgModule({
  declarations: [],
  imports: [
    CommonModule,
    TestRouterRoutingModule
  ]
})
export class TestRouterModule { }
"
`;

exports[`component generator should run successfully, finding a module 5`] = `
"import { ExampleComponent } from './example.component';
import { ExampleModule } from './example.module';
import { ComponentFixture, TestBed } from '@angular/core/testing';

describe('ExampleComponent', () => {
  let component: ExampleComponent;
  let fixture: ComponentFixture<ExampleComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [ExampleModule],
    });
    fixture = TestBed.createComponent(ExampleComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
"
`;

exports[`component generator should run successfully, when not including tests 1`] = `
"import { ExampleComponent } from './example.component';
import { ExampleModule } from './example.module';
import { Meta, moduleMetadata } from '@storybook/angular';

export default {
  id: 'examplecomponent-example',
  title: 'Components/Example',
  component: ExampleComponent,
  decorators: [
    moduleMetadata({
      imports: [ExampleModule],
    }),
  ],
} as Meta<ExampleComponent>;
export const Example = {
  render: (args: ExampleComponent) => ({
    props: args,
  }),
  args: {},
};
"
`;

exports[`component generator should run successfully, when not including tests 2`] = `
"import { Component } from '@angular/core';

@Component({
  selector: 'proj-example',
  templateUrl: './example.component.html',
  styleUrls: ['./example.component.scss'],
})
export class ExampleComponent {}
"
`;

exports[`component generator should run successfully, when not including tests 3`] = `
"import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ExampleRoutingModule } from './example-routing.module';
import { ExampleComponent } from './example.component';
@NgModule({
    declarations: [
        ExampleComponent
    ],
    imports: [
        CommonModule,
        ExampleRoutingModule
    ],
    exports: [ExampleComponent]
})
export class ExampleModule {
}
"
`;

exports[`component generator should run successfully, when not including tests 4`] = `
"import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { TestRouterRoutingModule } from './test-router-routing.module';


@NgModule({
  declarations: [],
  imports: [
    CommonModule,
    TestRouterRoutingModule
  ]
})
export class TestRouterModule { }
"
`;

exports[`component generator should use -storybook project 1`] = `
"import { ExampleComponent } from './example.component';
import { ExampleModule } from './example.module';
import { Meta, moduleMetadata } from '@storybook/angular';

export default {
  id: 'examplecomponent-example',
  title: 'Components/Example',
  component: ExampleComponent,
  decorators: [
    moduleMetadata({
      imports: [ExampleModule],
    }),
  ],
} as Meta<ExampleComponent>;
export const Example = {
  render: (args: ExampleComponent) => ({
    props: args,
  }),
  args: {},
};
"
`;

exports[`component generator should use -storybook project 2`] = `
"import { E2eVariations } from '@skyux-sdk/e2e-schematics';

describe('test-storybook', () => {
  E2eVariations.forEachTheme((theme) => {
    describe(\`in \${theme} theme\`, () => {
      beforeEach(() =>
        cy.visit(
          \`/iframe.html?globals=theme:\${theme}&id=examplecomponent-example--example\`
        )
      );
      it('should render the component', () => {
        cy.get('proj-example')
          .should('exist')
          .should('be.visible')
          .screenshot(\`examplecomponent-example--example-\${theme}\`)
          .percySnapshot(\`examplecomponent-example--example-\${theme}\`, {
            widths: E2eVariations.DISPLAY_WIDTHS,
          });
      });
    });
  });
});
"
`;
